src/spikeinterface/core/generate.py

```diff
@@ -317,6 +318,91 @@ def generate_snippets(
 
 
 ## spiketrain zone ##
+def synthesize_poisson_spike_vector(
+    num_units=20,
+    sampling_frequency=30000.0,
+    duration=60,
+    refractory_period_ms=4.0,
+    firing_rates=3.0,
+    seed=0,
+):
+    """
+    Generate random spike frames for neuronal units based on a Poisson process.
+
+    This function simulates the spike frames of a number of neuronal units, each firing according to
+    a Poisson process. The spike times are discretized to align with a given sampling frequency.
+
+    Parameters
+    ----------
+    num_units : int, optional
+        Number of neuronal units to simulate (default is 20).
+    sampling_frequency : float, optional
+        Sampling frequency in Hz (default is 30000.0).
+    duration : float, optional
+        Duration of the simulation in seconds (default is 60).
+    refractory_period_ms : float, optional
+        Refractory period between spikes in milliseconds (default is 4.0).
+    firing_rates : float or array_like, optional
+        Firing rate(s) in Hz. Can be a single value or an array of firing rates for each unit
+        (default is 3.0).
+    seed : int, optional
+        Seed for random number generator (default is 0).
+
+    Returns
+    -------
+    spike_frames : ndarray
+        1D array of spike frames.
+    unit_indices : ndarray
+        1D array of unit indices corresponding to each spike.
+
+    Notes
+    -----
+    - The function uses a geometric distribution to simulate the discrete inter-spike intervals,
+    based that would be an exponential process for continuous time.
```

Comment:
This isn't quite clear. Maybe it is missing a word. I'm not quite sure how to fix.

Reply:
You are correct this is not clear. I will expand on this. Thanks. Also for all the other comments. They all make sense and are very helpful as usual.

Reply:
I modified the docstring here, let me know what you think and if it is still unclear to you. Also, any other advice you might have is useful.